schemaless/create_uuid_map.py

```diff
@@ -16,100 +17,218 @@
+        pts_records = pts_helper.find_by_building_permit_number(permit_number)
+        if ppts_records:
+            # If there is a PPTS record that should be the parent
+            # record of PTS records with this permit number, we
+            # assign that as the parent.
+            parents.extend(ppts_records)
+        elif pts_records and pts_records[0] == fk:
+            # Even if there is no PPTS record that should be the
+            # parent of PTS records with this permit number, we need
+            # to ensure that all PTS records with the same permit number
+            # are assigned the same UUID.
+            #
+            # We do this by picking the first PTS record with the given
+            # permit number and assigning any other records with
+            # that permit number  as "children" of that record.
+            children.extend(pts_records[1:])
+
+
+class TCOHelper(RecordGraphBuilderHelper):
+    def process(self, fk, record, parents, children):
+        if 'building_permit_number' in record:
+            pts_helper = self.graph_builder.helpers[PTS.NAME]
+            parent_fk = pts_helper.find_by_building_permit_number(
+                record['building_permit_number'])
+            if parent_fk:
+                parents.extend(parent_fk)
+
+
+class MOHCDPipelineHelper(RecordGraphBuilderHelper):
+    def __init__(self, graph_builder):
+        super().__init__(graph_builder)
+        self._mohcd_id_to_fk = {}
+
+    def find_by_id(self, project_id):
+        """Find a fk by MOHCD project_id."""
+        return self._mohcd_id_to_fk.get(project_id, None)
+
+    def preprocess(self, latest_records):
+        for fk, record in latest_records.get(MOHCDPipeline.NAME, {}).items():
+            self._mohcd_id_to_fk[record['project_id']] = fk
+
+    def process(self, fk, record, parents, children):
+        if 'planning_case_number' in record:
+            ppts_helper = self.graph_builder.helpers[PPTS.NAME]
+            for parent in record['planning_case_number'].split(","):
+                parent_fk = ppts_helper.find_by_id(parent)
+                if parent_fk:
+                    parents.append(parent_fk)
+
 
-        mohcd_id_to_fk = {}
+class MOHCDInclusionaryHelper(RecordGraphBuilderHelper):
+    def __init__(self, graph_builder):
+        super().__init__(graph_builder)
+        self._mohcd_id_to_fk = {}
+
+    def find_by_id(self, project_id):
+        """Find a fk by MOHCD project_id."""
+        return self._mohcd_id_to_fk.get(project_id, None)
+
+    def preprocess(self, latest_records):
         for fk, record in latest_records.get(MOHCDPipeline.NAME, {}).items():
-            mohcd_id_to_fk[record['project_id']] = fk
+            self._mohcd_id_to_fk[record['project_id']] = fk
+
+    def process(self, fk, record, parents, children):
+        mohcd_pipeline_helper = self.graph_builder.helpers[
+            MOHCDPipeline.NAME]
+        mohcd_pipeline_helper.process(fk, record, parents, children)
```

Comment:
Why do we need to process the pipeline's helper here?

Reply:
After talking offline I was going to copy/paste the relevant code from MOHCDPipelineHelper, but ended up needing to have a reference to it anyway, so I kept this here and added a docstring.

Reply:
Approving, just a nit that its probably just better to be extra safe and copy paste the code to avoid things being confusing if they randomly changes the 'planning_case_number' field name

Reply:
done in 25039dd0998cba02dcf560668eedc7384f7bb8a7